src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,45 +1,101 @@
-import React from 'react';
+import { FC, useEffect, useState } from 'react';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  todo: Todo;
+  onClose: () => void;
+};
+
+export const TodoModal: FC<Props> = ({ todo, onClose }) => {
+  const {
+    id,
+    completed,
+    title,
+    userId,
+  } = todo;
+
+  const [user, setUser] = useState<User>();
+  const [isLoaded, setLoaded] = useState(false);
+  const [hasError, setError] = useState(false);
+
+  useEffect(() => {
+    setLoaded(false);
+    setError(false);
+
+    const getUserFromServer = async () => {
+      try {
+        const newUser = await getUser(userId);
+
+        setUser(newUser);
+      } catch {
+        setError(true);
+      } finally {
+        setLoaded(true);
+      }
+    };
+
+    getUserFromServer();
+  }, []);
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {!isLoaded && (
         <Loader />
-      ) : (
+      )}
+
+      {hasError && (
+        <div className="notification is-danger">
+          <button
+            type="button"
+            className="delete"
+            aria-label="close modal window"
+            onClick={onClose}
+          />
+
+          Something went wrong! Unable to load user. Try again.
+        </div>
+      )}
+
+      {user && (
         <div className="modal-card">
           <header className="modal-card-head">
             <div
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {`Todo #${id}`}
             </div>
 
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
             <button
               type="button"
               className="delete"
               data-cy="modal-close"
+              aria-label="close modal window"
+              onClick={onClose}
             />
           </header>
 
           <div className="modal-card-body">
             <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
+              {title}
             </p>
 
             <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
+              {completed
+                ? <strong className="has-text-success">Done</strong>
+                : <strong className="has-text-danger">Planned</strong>}
```

Comment:
You can use ternary only for value inside strong tag. Use classNames to set proper class.